lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4341,6 +4342,18 @@ namespace {
     }
 
     Decl *VisitFieldDecl(const clang::FieldDecl *decl) {
+      if (decl->hasAttr<clang::NoUniqueAddressAttr>()) {
+        if (const auto *rd = decl->getType()->getAsRecordDecl()) {
+          // Clang can store the next field in the padding of this one. Swift
+          // does not support this yet so let's not import the field and
+          // represent it with an opaque blob in codegen.
+          const auto &fieldLayout =
+              decl->getASTContext().getASTRecordLayout(rd);
+          if (!decl->isZeroSize(decl->getASTContext()) &&
+              fieldLayout.getDataSize() != fieldLayout.getSize())
+            return nullptr;
```

Comment:
What happens to derived conformances that rely on be presence of this field, eg Sendable or Equatable?

Reply:
Good point. Unfortunately, we would not consider these fields for `Sendable`. Maybe we could conservatively always say types with unimported fields should always be not `Sendable`? We already have many fields that we do not import for one reason or another.

Reply:
I think `Equatable` might already be really problematic for C++ types. We do not generate metadata for private fields, so I think even if we can auto conform a type to that the generated implementation is probably faulty :/

Reply:
It is not a problem for `Equatable`. Trying to conform a C++ the type to it gives this error message: `"error: extension outside of file declaring struct 'ReuseFieldPadding' prevents automatic synthesis of '==' for protocol 'Equatable'"`. Unfortunately, I don't think we can do anything about the Sendable problem for now.